src/crypto/hsm/nxp/PersistentStorageOperationalKeystoreHSM.cpp

```diff
@@ -0,0 +1,279 @@
+        ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> mPendingKeypair found");
+        return true;
+    }
+
+    while (i++ < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        if (keyidFabIdMapping[i].fabricIndex == fabricIndex)
+        {
+            ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> stored keyPair found");
+            return true;
+        }
+    }
+
+    ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> No key found");
+    return false;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::NewOpKeypairForFabric(FabricIndex fabricIndex,
+                                                                          MutableByteSpan & outCertificateSigningRequest)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    uint8_t slotId = getEmpytSlotId();
+
+    VerifyOrReturnError(slotId != 0, CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    ChipLogProgress(Crypto, "SE05x: New Op Keypair for Fabric %02x", fabricIndex);
+
+    // Replace previous pending keypair, if any was previously allocated
+    ResetPendingKey();
+
+    mPendingKeypair = Platform::New<Crypto::P256KeypairHSM>();
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_NO_MEMORY);
+
+    // Key id is created as slotid + start offset of ops key id
+    mPendingKeypair->SetKeyId(FABRIC_SE05X_KEYID_START + slotId);
+
+    err = mPendingKeypair->Initialize();
+    VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_NO_MEMORY);
+
+    mPendingFabricIndex = fabricIndex;
+
+    keyidFabIdMapping[slotId].isPending = 1;
+
+    size_t csrLength = outCertificateSigningRequest.size();
+    err              = mPendingKeypair->NewCertificateSigningRequest(outCertificateSigningRequest.data(), csrLength);
+    if (err != CHIP_NO_ERROR)
+    {
+        ResetPendingKey();
+        return err;
+    }
+    outCertificateSigningRequest.reduce_size(csrLength);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::ActivateOpKeypairForFabric(FabricIndex fabricIndex,
+                                                                               const Crypto::P256PublicKey & nocPublicKey)
+{
+    ChipLogProgress(Crypto, "SE05x: ActivateOpKeypair for Fabric %02x", fabricIndex);
+
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex) && (fabricIndex == mPendingFabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    // Validate public key being activated matches last generated pending keypair
+    VerifyOrReturnError(mPendingKeypair->Pubkey().Matches(nocPublicKey), CHIP_ERROR_INVALID_PUBLIC_KEY);
+
+    mIsPendingKeypairActive = true;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::CommitOpKeypairForFabric(FabricIndex fabricIndex)
+{
+    uint8_t i       = 0;
+    uint32_t slotId = mPendingKeypair->GetKeyId() - FABRIC_SE05X_KEYID_START;
```

Comment:
Shouldn't this be _after_ the mPendingKeypair null-check?

Reply:
Fixed this.